pool/cache.go

```diff
@@ -0,0 +1,47 @@
+package pool
+
+import (
+	"strings"
+
+	"github.com/bluele/gcache"
+	"github.com/nspcc-dev/neofs-api-go/pkg/session"
+)
+
+type SessionCache struct {
+	cache gcache.Cache
+}
+
+func NewCache() *SessionCache {
+	return &SessionCache{
+		cache: gcache.New(100).Build(),
+	}
+}
+
+func (c *SessionCache) Get(key string) *session.Token {
+	tokenRaw, err := c.cache.Get(key)
+	if err != nil {
+		return nil
+	}
+	token, ok := tokenRaw.(*session.Token)
+	if !ok {
+		return nil
+	}
```

Comment:
Can this condition ever be true? I think we can safely `return tokenRaw.(*session.Token)` from here.